### PR TITLE
chore: implement template and parameters for static site

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/static_site_test.go [hunk 1]
```diff
@@ -105,6 +107,142 @@ func TestStaticSite_NewStaticSite(t *testing.T) {
...
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			testManifest := manifest.NewStaticSite("frontend")
+
+			// GIVEN
+			conf := &StaticSite{
+				wkld: &wkld{
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/static_site_test.go]: any chance we can use the constructor here 🥺

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/deploy/cloudformation/stack/lb_web_svc.go [hunk 2]
```diff
@@ -142,9 +142,6 @@ func (s *LoadBalancedWebService) Template() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("convert the sidecar configuration for service %s: %w", s.name, err)
 	}
-	if err != nil {
```
Comment (paragbhingre) [inline internal/pkg/deploy/cloudformation/stack/lb_web_svc.go:145]: Thanks for removing this ❤️

File: internal/pkg/deploy/cloudformation/stack/static_site.go [hunk 3]
```diff
@@ -63,12 +65,62 @@ func NewStaticSite(conf *StaticSiteConfig) (*StaticSite, error) {
...
+		NestedStack:         addonsOutputs,
+		PermissionsBoundary: s.permBound,
+
+		// Custom Resource Config.
+		CustomResources: crs,
+	})
+	if err != nil {
+		return "", err
+	}
+	return content.String(), nil
```
Comment (paragbhingre) [inline internal/pkg/deploy/cloudformation/stack/static_site.go:101]: Would there be any task def overrides before this?

Comment (iamhopaul123) [reply internal/pkg/deploy/cloudformation/stack/static_site.go:101]: Task def override is only for ECS service. `static site` service type won't have an ECS service.

**Review by paragbhingre — APPROVED** [review]

**Pull request merged**